rts/Sim/Units/Scripts/UnitScript.cpp

```diff
@@ -583,7 +587,20 @@ bool CUnitScript::EmitAbsSFX(int sfxType, const float3& absPos, const float3& ab
 				w->weaponMuzzlePos = origWeaponMuzzlePos;
 
 				const bool origRestored = w->Attack(origTarget);
-				assert(origRestored);
+				// originally this one used to follow, but it's incorrect
+				// the w->Fire(true); might have killed the same original target
+				// origRestored is then false and the emitTarget remains the target
+				// which will cause shooting a phantom, given the
+				// emitTarget used to be {absPos + absDir} and absDir is one elmo sized,
+				// the unit could also move, which would cause the weapon to shoot in
+				// "random directions" (in fact to emitTarget, which is absurdly close to
+				// absPos.
+				//assert(origRestored);
```

Comment:
is there any value keeping a description of some old and fixed bug? how about just

```cpp
const bool origRestored = w->Attack(origTarget);

// can be false if w->Fire killed our target
if (!origRestored)
  w->DropCurrentTarget();
```